hrms/hr/doctype/job_opening/job_opening.py

```diff
@@ -73,38 +113,17 @@ def update_job_requisition_status(self):
 			job_requisition.save()
 
 	def get_context(self, context):
+		job_applicants = frappe.db.count("Job Applicant", {"job_title": self.name})
+		context.no_of_applications = job_applicants
 		context.parents = [{"route": "jobs", "title": _("All Jobs")}]
+		context.posted_on = pretty_date(self.posted_on)
 
 
-def get_list_context(context):
-	context.title = _("Jobs")
-	context.introduction = _("Current Job Openings")
-	context.get_list = get_job_openings
-
-
-def get_job_openings(
-	doctype, txt=None, filters=None, limit_start=0, limit_page_length=20, order_by=None
-):
-	fields = [
-		"name",
-		"status",
-		"job_title",
-		"description",
-		"publish_salary_range",
-		"lower_range",
-		"upper_range",
-		"currency",
-		"job_application_route",
-	]
-
-	filters = filters or {}
-	filters.update({"status": "Open"})
-
-	if txt:
-		filters.update(
-			{"job_title": ["like", "%{0}%".format(txt)], "description": ["like", "%{0}%".format(txt)]}
-		)
-
-	return frappe.get_all(
-		doctype, filters, fields, start=limit_start, page_length=limit_page_length, order_by=order_by
-	)
+def close_expired_job_openings():
+	today = getdate()
+	for d in frappe.get_all(
+		"Job Opening",
+		filters={"Status": "Open", "closes_on": ["<=", today]},
+		fields=["name", "closes_on"],
+	):
+		frappe.set_value("Job Opening", d.name, "status", "Closed")
```

Comment:
```suggestion
		doc = frappe.get_doc("Job Opening", d.name)
		doc.status = "Closed"
		doc.flags.ignore_permissions = True
		doc.flags.ignore_mandatory = True
		doc.save()
```

Let's do a doc update instead of a db update. Users might set up notifications on openings when they auto-close. These won't be triggered if we do a direct db update